frontend/src/Components/Supplies/Supplies.js

```diff
@@ -124,21 +132,23 @@ class Supplies extends React.Component {
         })
     }
 
-    onClickDeleteSelected = (rowsDeleted) => {
+    onClickDeleteSelected = () => {
+        let {data, rowsSelected} = this.state;
         let allOk = true;
         let someOk = false;
-        for (let key in rowsDeleted) {
-            deleteItem(this.state.data[key].id)
+        for (let key in rowsSelected) {
+            let supplyId = data[rowsSelected[key].index].id;
+            deleteItem(supplyId)
                 .then((res) => {
                     someOk = true;
+                    this.updateData();
                 }).catch((err) => {
                     console.error(err);
                     allOk = false;
                 });
         }
 
-        this.updateData();
-
+    
         if (allOk) {
```

Comment:
same as before, deleteItem is async function. allOk may be set to false after you display snackbar.